ENG-508: Create a Load and Soak testing framework

musitdev opened the pull request:
# Summary
- **Status**: `draft` <!-- `review`, `suspended` -->
- **RFCs**: 
- **Categories**:  Utils Test

Develop mini framework for load and soak testing

Initial architecture definition:

```mermaid
flowchart TD
  subgraph Execute process
    PROCESS("process-compose:
     Entry point, 
     start all process")
  end
  subgraph Full Node
    SUZ(Suzuka Node)
    CEL(celestia)
    SEQ(Sequencer)
  end
  subgraph Tester
    SCH("Scheduler:
    - Schedule client execution.
    - Use rayon to schedule 
    - Each client in its own thread.
    - When a client execution end, return the result.
    - Store client execution info. Use formated logs" )
    CONF("Config:
    Define the test parameters")
    CLI("Client:
    - Execute several scenarios in async mode.
    - Log: scenario errors.
     ")
    SCE("Scenario:
    Define the scenario steps.")
  end

  PROCESS-- Start -->SUZ
  PROCESS-- Start -->CEL
  PROCESS-- Start -->SEQ
  PROCESS-- Start -->SCH
  CONF-- Define Exec -->SCH
  SCH-- Start Execution -->CLI
  CLI-- Run the scenario -->SCE
  SCE-- Return exec result -->CLI
  CLI-- Return exec result -->SCH
```
Scenario are developed in Rust and are linked with the Test framework. Scenario are defined in the bin folder of the crate.

The runtime start several clients that are executed using Rayon crate. Each client execute several scenarios using a mono thread Tokio async runtime.  For load test, the number of client is defined by the total number of scenario to start divided by the number of scenario per client, see ExecutionConfig. 

Execution data are written in a formatted (json) log file to be processed later if needed.
A function to log in this formatted file is provided in the scenario trait.

After the execution, the Test framework exe print on the stdout a final report.

I put the framework crate in `util/load-soak-testing`, see the changed files.

# Changelog

<!-- 
Describe your changes. List roughly in order of importance.
-->

# Testing

<!--
Describe your Test Plan and explain added or modified test components.
-->

# Outstanding issues
<!--
List any outstanding issues that need to be addressed in future PRs, but which do not block merging this PR.
-->

musitdev commented:
I don't know why I get a lot of previous commit. I've started from main and rebase it. If you've an idea.
The developed files are in util/load_soak_testing

l-monninger commented:
Overall this is good for now. But, please fix the typos and for organization, I would prefer to see the tests in the respective network or client crate instead of within the framework crate. I would prefer to include the framework where need and have the tests organized by service.

musitdev commented:
I move the crate in networks/suzuka/load-soak-testting
I've implemented a simple soak test that loop and execute scenario during the test duration.
I've corrected some typos mistake.
The remaining part is to run new scenarios from time to time in a cycling way on top of the base looping ones.

l-monninger commented on networks/suzuka/load-soak-testing/src/lib.rs:
```diff
@@ -0,0 +1,393 @@
+		}
+	}
+
+	pub fn calcualte_average_exec_time_milli(sceanarios: &[ScenarioExecMetric]) -> u128 {
```
There are still typos in here.

l-monninger commented on networks/suzuka/load-soak-testing/bin/basicalicebob.rs:
```diff
@@ -0,0 +1,190 @@
+use anyhow::{Context, Result};
+use aptos_sdk::{
+	coin_client::CoinClient,
```
For organization, I still want you to move these out into the appropriate network crates. `suzuka-client` for example.

l-monninger commented on networks/suzuka/load-soak-testing/bin/demoscenario.rs:
```diff
@@ -0,0 +1,59 @@
+use anyhow::Result;
```
Please fix typos and change organization before merging.

musitdev commented:
Recreate the PR in #111 to remove unknown modifications.
